syntax = "proto2";

message PBHeader {
  required uint32 magic_number = 1;
  required uint32 version_major = 2;
  required uint32 version_minor = 3;
}

message PBResultSet {
  message PBSensorData {
		optional bytes batterie_value = 1;
	}
	message PBPosition {
		optional double x_position = 1 [default = 1.79769e+308];
		optional double y_position = 2 [default = 1.79769e+308];
		optional double z_position = 3 [default = 1.79769e+308];
	}
	message PBRSSIEntry {
		required uint32 anchor_id = 1;
		required uint32 channel = 2;
		required uint32 rssi_value = 3;
	}

	required uint32 timestamp_sec = 1;
	required uint32 timestamp_usec = 2;
	required uint32 src_id = 3;
	required uint32 blink_id = 4;
	optional PBPosition position = 5;
	optional bytes payload = 6;
	optional PBSensorData sensor_data = 7;
	repeated PBRSSIEntry rssi_entry = 8;
}